tests/chainer_tests/test_variable.py

```diff
@@ -366,6 +366,67 @@ def test_addgrad_gpu_to_another_gpu(self):
             c = cp.full(3, 30, dtype=np.float32)
         self.check_addgrad(a, b, c)
 
+    def test_debug_print_cpu(self):
+        arr = np.random.randn(5, 3, 5, 5).astype(np.float32)
+        v = chainer.Variable(arr)
+
+        result = v.debug_print()
+        self.assertTrue(repr(v) in result)
+        self.assertTrue('device: CPU' in result)
+        self.assertTrue('volatile: OFF' in result)
+        self.assertTrue("numpy.ndarray" in result)
+        self.assertTrue("dtype: float32" in result)
+
+        msg = "statistics: mean={mean}, std={std}".format(mean=arr.mean(),
+                                                          std=arr.std())
+        self.assertTrue(msg in result)
+        self.assertTrue("grad: None" in result)
+
+        # zero grad
+        v.zerograd()
+        result = v.debug_print()
+        self.assertTrue("grad: 0" in result)
+
+        # add grad
+        grad = np.random.randn(5, 3, 5, 5).astype(np.float32)
+        v.grad = grad
+        result = v.debug_print()
+
+        msg = "grad: mean={mean}, std={std}".format(mean=grad.mean(),
+                                                    std=grad.std())
+        self.assertTrue(msg in result)
+
+    @attr.gpu
+    def test_debug_print_gpu(self):
+        arr = np.random.randn(5, 3, 5, 5).astype(np.float32)
+        v = chainer.Variable(arr)
+        v.to_gpu()
```

Comment:
I think it is better to explicitly send `v` to GPU 0 because default device can be other than 0 in some environment (although I am not certain such an environment does exist).